app/main.py

```diff
@@ -1,6 +1,16 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    memory = {}
+
+    @functools.wraps(func)
+    def inner(*args, **kwargs) -> dict:
+        if args in memory.keys():
```

Comment:
why are you checking only `args in memory` what if there are functions with the same positional arguments but with different keys

Reply:
I hope I understood correctly